File: Year-2023/LeetCode-February-Challenge/BestTimetoBuyandSellStock.js
var maxProfit = function(prices) {
    let minPrices = Infinity, maxProfit = -1;
    for(let x of prices){
        minPrices = Math.min(minPrices, x);
        maxProfit = Math.max(maxProfit, x - minPrices);
    }
    return maxProfit;
};

let prices = [7,1,5,3,6,4];
console.log(maxProfit(prices));

File: Year-2023/LeetCode-February-Challenge/BestTimetoBuyandSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE, maxProfit = Integer.MIN_VALUE;
        for(int i = 0; i < prices.length; ++i){
            minPrice = Math.min(minPrice, prices[i]);
            maxProfit = Math.max(maxProfit, prices[i] - minPrice);
        }
        return maxProfit;
    }
}
public class BestTimetoBuyandSellStock {
    public static void main(String[] args) {
        int[] prices = {7,1,5,3,6,4};
        Solution s= new Solution();
        System.out.println(s.maxProfit(prices));
    }
}